src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -318,6 +321,209 @@ private List<Diagnostic> GetAllDiagnostics(HttpRequestParseResult parseResult)
             return node.CreateBindingSuccess(value);
         }
 
-        return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        var bindingResult = MatchExpressionValue(node, expression);
+        if ( bindingResult is not null)
+        {
+            return bindingResult;
+        } else
+        {
+            return node.CreateBindingFailure(HttpDiagnostics.CannotResolveSymbol(variableName));
+        }
+
+        
     }
+
+    private HttpBindingResult<object?>? MatchExpressionValue(HttpExpressionNode node, string expression)
+    {
+
+        var guidPattern = new Regex(@$"\{"$guid"}", RegexOptions.Compiled);
+        var dateTimePattern = new Regex(@$"\{"$datetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var localDateTimePattern = new Regex(@$"\{"$localDatetime"}\s(?<type>rfc1123|iso8601|'.+'|"".+"")(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
+        var randomIntPattern = new Regex(@$"\{"$randomInt"}(?:\s(?<parameters>-?\d+)){{0,2}}", RegexOptions.Compiled);
+        var timestampPattern = new Regex(@$"\{"$timestamp"}(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?", RegexOptions.Compiled);
```

Comment:
Please extract a string constant for `(?:\s(?<offset>-?\d+)\s(?<option>y|M|Q|w|d|h|m|s|ms))?` since it is repeated in 3 of the regexes above.

Reply:
Same for `(?<type>rfc1123|iso8601|'.+'|"".+"")` which seems to be repeated in 2 places.